src/modules/vocabulary/vocabulary.service.ts

```diff
@@ -185,4 +186,54 @@ export class VocabularyService {
       isCorrect
     }
   }
+
+  async getVocabsByTopicId(
+    topicId: number,
+  ): Promise<VocabularyResponse[]> {
+
+    const data = await this.vocabularyRepo.find({
+      where: {
+        topic: { id: topicId },
+        isActive: true
+      },
+      order: { createdAt: "ASC" },
+    });
+
+    const result = data.map(v => {
+      return {
+        id: v.id,
+        word: v.word,
+        pronounce: v.pronounce,
+        partOfSpeech: v.partOfSpeech,
+        meaning: v.meaning,
+        exampleSentence: v.exampleSentence,
+      }
+    });
+
+    return result;
+  }
+
+  async getTestVocabulariesByTopic(
+    topicId: number,
+  ): Promise<TestVocabResponse[]> {
+
+    const data = await this.vocabularyRepo.find({
+      where: {
+        topic: { id: topicId },
+        isActive: true
```

Comment:
```suggestion
      where: {
        topicId: topicId,
        isActive: true
```

use topicId instead of topic relation to avoid JOIN for performance. There are many place do the same, please correct them all